camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -280,20 +285,10 @@ private void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSour
 
     List<Properties> jobPropsList = planner.createSweeperJobProps(topic, topicSourceDir, topicDestDir, fs);
 
-    for (Properties jobProps : jobPropsList) {
+    for (Properties jobProps : jobPropsList){
       tasksToComplete.add(runCollector(jobProps, topic));
     }
 
-    for (Future<?> task : tasksToComplete) {
-      try {
-        task.get();
-      } catch (InterruptedException e) {
-        throw e;
-      } catch (ExecutionException e) {
-        throw e;
-      }
-    }
-
```

Comment:
I didn't write this code, but reading documentation of `ExecutorService` really helps to understand. So the flow is as follows: `shutdown`, `while(!isNotTerminated)`, `awaitTermination`. Documentation of `isNotTerminated` is as follows:

> Returns true if all tasks have completed following shut down.

So we will be waiting for all of them to finish.